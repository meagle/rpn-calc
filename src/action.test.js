// @flow
import {
  addInputToStack,
  removeFromStack,
  sendOperandToStack,
  sendOperatorToStack,
} from './actions';

test('Sends an operand to the stack', () => {
  const expected = {
    type: 'USER_NUMERIC_INPUT',
    key: {
      operand: '5',
    },
  };

  const actual = sendOperandToStack({
    operand: '5',
  });
  expect(actual).toEqual(expected);
});

test('Sends an operator to the stack', () => {
  const actual = sendOperatorToStack({
    arity: 1,
    operator: 'sqrt',
    fn: n => {
      return Math.sqrt(n);
    },
  });
  expect(actual).toMatchSnapshot();
});

test('Adds the current input to the stack', () => {
  const expected = {
    type: 'ADD_TO_STACK',
    value: 42,
  };
  const actual = addInputToStack(42);
  expect(actual).toEqual(expected);

  const actual2 = addInputToStack('42');
  expect(actual2).toEqual(expected);
});

test('Removes from the stack or current input', () => {
  const expected = {
    type: 'REMOVE_FROM_STACK',
    userInput: '42',
  };
  const actual = removeFromStack('42');
  expect(actual).toEqual(expected);
});

test('Removes from the stack when there is no current input', () => {
  const expected = {
    type: 'REMOVE_FROM_STACK',
  };
  const actual = removeFromStack();
  expect(actual).toEqual(expected);
});
