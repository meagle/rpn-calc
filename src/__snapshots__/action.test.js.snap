// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sends an operator to the stack 1`] = `
Object {
  "key": Object {
    "arity": 1,
    "fn": [Function],
    "keyValue": "s",
    "operator": "sqrt",
  },
  "type": "USER_OPERATOR_INPUT",
}
`;
